cpp/include/cudf/utilities/type_dispatcher.hpp

```diff
@@ -163,6 +166,9 @@ MAP_TIMESTAMP_SCALAR(timestamp_ns)
 template <typename T>
 using scalar_type_t = typename type_to_scalar_type_impl<T>::ScalarType;
 
+template <typename T>
+using scalar_device_type_t = typename type_to_scalar_type_impl<T>::ScalarDeviceType;
```

Comment:
`type_t` is a bit redundant, since `_t` means type.  I would rename this to `scalar_device_t` and above to `scalar_t`.

Reply:
I was copying what was already there.  Changing this (at least the scalar_type_t) would require changing all references.  I can do that if that's what we want.